### PR TITLE
 Refactor `toast` component to use theme context

### PR DESCRIPTION
**Pull request by smohamedjavid**
### Summary

This PR refactors the `Toast` component to use theme context for consistency across the codebase. 

Each toast will have its own `theme-provider` which will help the children to use the context properly without interfering with other toast's context. 

Below is an example of providing a specific theme for toast.

```clojure
#(rf/dispatch
   [:toasts/upsert
    {:icon        :i/info
     :icon-color  colors/danger-50
     :theme       :dark ;; If it's nil (or the key is not present), it will fall back to the system theme.
     :text        (i18n/label :t/camera-permission-denied)}])}]))
```

#### Following components are refactored to use `with-theme` / `theme-context`:

- [x] `text` component
- [x] `user-avatar` component
- [x] `icon` component
- [x] `circle-timer` component

This PR also updates the icon keyword used in the `:toasts/upsert` event dispatch to respect icon [guidelines](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#icons).

### Testing notes

- Please test whether all the Toasts across the application are working as expected on both light mode and dark mode.

- Please test the Toasts in Syncing process as it is dark-themed. It should not change depending on the user's theme.

If any issue is encountered, kindly check whether the issue is reproducible with nightly or the latest `develop` branch. This would help us to identify issues due to changes in this PR and resolve them.

### Platforms

- Android
- iOS

status: ready

### CODE REVIEW

File: src/status_im2/common/home/actions/view.cljs [hunk 1]
```diff
@@ -215,7 +215,7 @@
                                  (rf/dispatch [:hide-bottom-sheet])
                                  (rf/dispatch [:toasts/upsert
                                                {:id         :remove-nickname
-                                                :icon       :correct
+                                                :icon       :i/correct
```
Comment (J-Son89) [inline src/status_im2/common/home/actions/view.cljs:220]: 👍

**Review by J-Son89 — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Review by yqrashawn — APPROVED** [review]

**Comment by smohamedjavid** [comment]
Marking as draft as we will need to update text, and icon components to use theme context (those styles are overriding from toast), and update toast with a theme provider.

File: src/quo2/components/notifications/toast/view.cljs [hunk 2]
```diff
@@ -38,42 +45,47 @@
...
   [{:keys [icon icon-color title text action undo-duration undo-on-press container-style
-           override-theme user]}]
-  [toast-container
-   {:left            (cond icon
-                           [icon/icon icon
-                            (cond-> (style/icon override-theme)
-                              icon-color
-                              (assoc :color icon-color))]
+           theme user]}]
+  (let [context-theme (or theme (quo.theme/get-theme))]
```
Comment (J-Son89) [inline src/quo2/components/notifications/toast/view.cljs:73]: is this a temporary solution while we update uses across the codebase? 🤔

Comment (smohamedjavid) [reply src/quo2/components/notifications/toast/view.cljs:73]: No, That's for the theme provider below that line.
Fallback to user-theme if a theme is not specified.

It's similar to what we do with screens or bottom sheets.
https://github.com/status-im/status-mobile/blob/33d41dc5d55b6e976edaf23ac2caf6ab3d1c87bc/src/status_im2/navigation/view.cljs#L66

Comment (J-Son89) [reply src/quo2/components/notifications/toast/view.cljs:73]: yeah you are right, maybe we should make it a `sub` instead? 🤔  across all places

Comment (J-Son89) [reply src/quo2/components/notifications/toast/view.cljs:73]: anyway it's beyond this pr 👍

**Comment by VolodLytvynenko** [comment]
hi @smohamedjavid thank you for PR. Take a look please, one issue if found
## ISSUE 1: [Android] The highlighted background around the text is shown on the toast when dark mode is used

### Steps:
1. Set up dark mode within the app
2. Check the toast

### Actual result:
The highlighted background around the text is shown (it is slightly visible)
image

### Expected result:
Last nightly
image

### Devices:
- Android studio emulator: Pixel 6 API 31. v12
- Real device: Pixel 7a, Android 13

**Review by ilmotta — APPROVED** [review]
Nice PR @smohamedjavid

**Comment by smohamedjavid** [comment]
@VolodLytvynenko - Thanks for testing the PR!

> ISSUE 1: [Android] The highlighted background around the text is shown on the toast when dark mode is used

The issue is fixed. Kindly retest.

**Comment by VolodLytvynenko** [comment]
Hi @smohamedjavid thank you for fixes. No issues from my side. PR is ready to be merged

**Pull request merged**